### ISSUE TITLE
doom upgrade: Open diff in browser -> Bad file descriptor

### ISSUE THREAD

**Issue by fosskers** [hlissner/doom-emacs#5324]
### What did you expect to happen?

When I run `doom upgrade` and agree to view the diff in my browser, I shouldn't see a strange warning about bad file descriptors.

### What actually happened?

Upon `doom upgrade` I'm asked if I'd like to see the diff in my browser. **If I have a browser open** it always sends me an error message that looks like this:

![bad-file-desc](https://user-images.githubusercontent.com/229679/127757178-622c4c72-09d6-4e2b-926c-4b566cd4504d.png)

Although the page opens successfully in the browser anyway.

**If I never had a browser open in the first place** I see no such error.

### Steps to reproduce

1. Have Emacs open and Doom on a non-recent commit.
2. Have a browser open (Firefox?)
3. Run `doom upgrade` (from a `vterm` is how I've been doing it).
4. Agree to view the diff.
5. See the error message.

### System Information

https://pastebin.com/NVpjkfwA

**Comment by destroyer449** [comment]
I get the same error on every upgrade, but I can still proceed with the upgrade, so I simply ignore it.

**Comment by fosskers** [comment]
Yes I should mention that this doesn't actually affect the ability to upgrade.

**Comment by destroyer449** [comment]
I'm not on my laptop to test it right now, but I do remember that this issue has existed for at least a couple of months. Perhaps someone could `git bisect` to find where it began?
